workflow/controller/operator.go

```diff
@@ -1202,18 +1202,21 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 
 		// we only need to update these values if the container transitions to complete
 		if newPhase.Fulfilled() {
+			exitCode := getExitCode(pod)
 			// outputs are mixed between the annotation (parameters, artifacts, and result) and the pod's status (exit code)
-			if exitCode := getExitCode(pod); exitCode != nil {
-				woc.log.Infof("Updating node %s exit code %d", node.ID, *exitCode)
-				node.Outputs = &wfv1.Outputs{ExitCode: pointer.StringPtr(fmt.Sprintf("%d", int(*exitCode)))}
-				if outputStr, ok := pod.Annotations[common.AnnotationKeyOutputs]; ok {
-					woc.log.Infof("Setting node %v outputs: %s", node.ID, outputStr)
-					if err := json.Unmarshal([]byte(outputStr), node.Outputs); err != nil { // I don't expect an error to ever happen in production
-						node.Phase = wfv1.NodeError
-						node.Message = err.Error()
-					}
+			if exitCode != nil || pod.Status.Reason == common.ErrDeadlineExceeded {
```

Comment:
I think this condition is not needed. All that matters is that we have the annotation. I don't think we need to guard it. Try removing replacing the guard based on exit code with a guard based on `outputStr,ok := pod.Annotations[common.AnnotationKeyOutputs]; ok`